### DISCUSSION TITLE
Simulating AGB-WD binary system

### DISCUSSION THREAD

**Discussion by JamAtGit** [AMReX-Astro/Castro#2712] [General]
Hi everyone, I would like to simulate a binary system formed by an AGB star and a compact companion (WD), and I wonder how to  setup this problem in Castro .. Any idea to start from?

Thanks

Regards

Jose

**Comment by zingale** [comment]
You can find an example of two white dwarfs in `Castro/Exec/science/wdmerger`.  In that setup, the WD models are created internally.  If you instead want to use models that were created by a stellar evolution code, you should first remap them to a uniform grid and make them consistent with our EOS and discrete form of HSE.  An example of doing this for a massive star is here:
https://github.com/AMReX-Astro/initial_models/tree/main/massive_star

You can then read in the models and interpolate to the grid.  There are lots of examples that do this for a single model, like the `subchandra`  setup.

The model reader does support multiple models, so it should not be hard to have the initialization read in two models, and then you would have two decide which model to map from in `problem_initialize_state_data.H`.
